docs/customization.rst

```diff
@@ -174,101 +197,257 @@ Configuration Options
-        to use this feature.
+        A list of dictionaries used to populate the :ref:`version_dropdown` selector.
 
-    .. confval:: version_dropdown_text
+.. _version_dropdown:
 
-        The text in the version dropdown button
+Version Dropdown
+================
 
-    .. confval:: version_json
+A version dropdown selector is available that lets you store multiple versions in a single site.
+The standard structure of the site (relative to the base domain) is usually
 
-        The location of the JSON file that contains the version information. The default assumes there
-        is a file versions.json located in the root of the site.
+.. code-block:: text
 
-    .. confval:: version_info
+    /
+    /1.0
+    /2.0
+
+Whereas Sphinx must be executed seperately for each version of the documentation you are
+building.
+
+.. code-block:: text
+
+    git fetch --all --tags
+    git checkout 1.0
+    sphinx-build docs _build/1.0
+    git checkout 2.0
+    sphinx-build docs _build/2.0
+
+This means you need to implement a custom mechanism that allows keeping the built
+documentation when checking out different branches or tagged commits.
+
+.. important::
+    To use the version dropdown selector, you must set :confval:`version_dropdown` to `True` in
+    the `html_theme_options` dict.
+
+    .. code-block:: python
+
+        html_theme_options = {
+            "version_dropdown": True,
+        }
+
+Supported Approaches
+********************
+
+There are two approaches:
+
+1. The version information is stored in a :confval:`version_info` `list` in the conf.py file.
+   
+   .. note::
+       Notice this is an ordered list. Meaning, aproach 1 will take precedence over approach 2.
+
+2. The version information is stored in a JSON file.
+   
+   The default name of the JSON file is ``versions.json``, but the JSON file's name could be
+   changed by setting :confval:`version_json` in the conf.py file's `html_theme_options`.
+
+   .. code-block:: python
+
+       html_theme_options = {
+           "version_json": "doc_versions.json",
+       }
+  
+   .. warning::
+       The JSON approach only works if your documentation is served from a webserver; it does not
+       work if you use ``file://`` url). When serving the docs from a webserver the
+       :confval:`version_json` file is resolved relative to the *parent* directory that contains
+       the sphinx builder's HTML output. For example, if the builder's output is ``2.0``, you
+       should have directory structure like so:
+   
+       .. code-block:: text
+   
+           /versions.json
+           /1.0/index.html
+           /2.0/index.html
+
+
+Version Information Structure
+*****************************
+
+Both approaches use a data structure similar to what is used by the
+`mkdocs mike plugin <https://github.com/jimporter/mike>`_. Contrary to what the mike plugin's
+README says, the ``alias`` field is not optional, but it can be set to an empty list if not using
```

Comment:
Should be `aliases`, also we should perhaps document the meaning of `aliases`.

Reply:
Agreed. The mike plugin's README doesn't seem to explain the alias in detail (other than exemplifying it's usage). I'm guessing you know how to use this better after modifying the JS for this theme.